x-pack/plugins/data_enhanced/public/search/search_interceptor.ts

```diff
@@ -46,9 +54,21 @@ export class EnhancedSearchInterceptor extends SearchInterceptor {
       : TimeoutErrorMode.CONTACT;
   }
 
+  private createRequestHash$(request: IKibanaSearchRequest, options: IAsyncSearchOptions) {
+    const { sessionId } = options;
+    const { preference, ...params } = request.params || {};
+    const hashOptions = {
+      ...params,
+      sessionId,
```

Comment:
I think we want to pass all options? 
Because, for example, I think we don't want: 

```
{
 sessionId: 'id',
 isRestore: true
}
```

to cache hit: 

```
{
 sessionId: 'id',
 isRestore: true
}
```